x/likenft/abci.go

```diff
@@ -0,0 +1,38 @@
+package likenft
+
+import (
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/likecoin/likechain/x/likenft/keeper"
+	"github.com/likecoin/likechain/x/likenft/types"
+)
+
+// EndBlocker called every block, process class reveal queue.
+func EndBlocker(ctx sdk.Context, keeper keeper.Keeper) {
+	defer telemetry.ModuleMeasureSince(types.ModuleName, time.Now(), telemetry.MetricKeyEndBlocker)
+
+	logger := keeper.Logger(ctx)
+
+	// Reveal class when its reveal time is reached.
+	keeper.IterateClassRevealQueue(ctx, func(classId string) bool {
+		// Note: Create public functions from keeper files and access them here instead of mutating directly
+
+		// TODO: Mint unclaimed tokens supply to class owner #206
+
+		// TODO: Reveal nft class #206
+
+		// TODO: Remove class from queue #206
+
+		ctx.EventManager().EmitTypedEvent(&types.EventRevealClass{
+			ClassId: classId,
+		})
+
```

Comment:
btw, can we see how to handle exceptions? like where to log & make sure errors will only cause the entry to be skipped and not revert the whole batch?

on exceptions, how to let user trigger retry?

Reply:
errors should be logged by the logger and the entry shall be retried when the next block arrives if it's not removed from the queue

Reply:
- let's add success status & error message at the reveal event
- we should dequeue if failed, and consider if we need to expose a manual reveal method (depending on the failure modes, after we implement the logic)
- i suppose the logic will return a error, but can we capture all unhandled panic / exceptions here?